storagenode/console/consoleserver/server.go

```diff
@@ -79,166 +102,234 @@ func NewServer(logger *zap.Logger, config Config, service *console.Service, list
+	walletAddress := server.service.GetWalletAddress(ctx)
 
-	versionInfo := s.service.GetVersion(ctx)
+	versionInfo := server.service.GetVersion(ctx)
 
-	err = s.service.CheckVersion(ctx)
+	err = server.service.CheckVersion(ctx)
 	if err != nil {
-		s.log.Error("can not check latest storage node version", zap.Error(err))
-		response.Error = "can not check latest storage node version"
-		w.WriteHeader(http.StatusInternalServerError)
-		return
+		return response, err
 	}
 
-	uptime := s.service.GetUptime(ctx)
-	nodeID := s.service.GetNodeID(ctx)
+	bandwidthChartData, err := server.getBandwidthChartData(ctx, satelliteID)
+	if err != nil {
+		return response, err
+	}
 
-	response.Data.DiskSpace = *space
-	response.Data.Bandwidth = *usage
-	response.Data.WalletAddress = walletAddress
-	response.Data.VersionInfo = versionInfo
-	response.Data.IsLastVersion = true
-	response.Data.Uptime = uptime
-	response.Data.NodeID = nodeID.String()
-	response.Data.Satellites = satellites
+	// TODO: uncomment in future, when caching will be implemented
+	// diskSpaceChartData, err := server.getDiskSpaceChartData(ctx, satelliteID, satellites)
+	// if err != nil {
+	// 	return response, err
+	// }
+
+	uptime := server.service.GetUptime(ctx)
+	nodeID := server.service.GetNodeID(ctx)
+
+	// TODO: uncomment in future, when caching will be implemented
+	// if satelliteID != nil {
+	// 	satelliteStats, err := server.service.GetStatsFromSatellite(ctx, *satelliteID)
+	// 	if err != nil {
+	// 		return response, err
+	// 	}
+	//
+	// 	response.UptimeCheck = satelliteStats.UptimeCheck
+	// 	response.AuditCheck = satelliteStats.AuditCheck
+	// }
+
+	response.DiskSpace = *space
+	response.Bandwidth = *usage
+	response.WalletAddress = walletAddress
+	response.VersionInfo = versionInfo
+	response.IsLastVersion = true
+	response.Uptime = uptime
+	response.NodeID = nodeID.String()
+	response.Satellites = satellites
+	response.BandwidthChartData = bandwidthChartData
+	//response.DiskSpaceChartData = diskSpaceChartData
+
+	return response, nil
+}
 
-	w.WriteHeader(http.StatusOK)
+func (server *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
+	if satelliteID != nil {
+		return server.service.GetBandwidthBySatellite(ctx, *satelliteID)
+	}
+
+	return server.service.GetUsedBandwidthTotal(ctx)
+}
+
+func (server *Server) getBandwidthChartData(ctx context.Context, satelliteID *storj.NodeID) (_ []console.BandwidthUsed, err error) {
+	from, to := getMonthRange()
+
+	if satelliteID != nil {
+		return server.service.GetDailyBandwidthUsed(ctx, *satelliteID, from, to)
+	}
+
+	return server.service.GetDailyTotalBandwidthUsed(ctx, from, to)
 }
 
-func (s *Server) getBandwidth(ctx context.Context, satelliteID *storj.NodeID) (_ *console.BandwidthInfo, err error) {
+func (server *Server) getDiskSpaceChartData(ctx context.Context, satelliteID *storj.NodeID, satellitesID storj.NodeIDList) (result []nodestats.SpaceUsageStamp, err error) {
```

Comment:
This satelliteID should not be a pointer

Reply:
Also this method is hard to read. Is it possible to split this into two functions?

Reply:
> This satelliteID should not be a pointer

using it as a pointer here allows me easily to understand should i fetch data from all satellites or from one concrete